runtime/eventlog.c

```diff
@@ -41,32 +41,80 @@
+/* Description of a user event for the 'metadata' stream */
+const char_os *user_event_format = "\nevent {\n     id = %d;\n     name = \"%s\";\n     stream_id = 0;\n     fields := struct {\n        uint8_t span_type;\n     };\n};\n";
+static struct ctf_metadata_header metadata_header = {
+  CTF_METADATA_MAGIC,
+  CTF_TRACE_UUID,
+  0,
+  0,
+  0,
+  0,
+  0,
+  0,
+  CTF_MAJOR_VERSION,
+  CTF_MINOR_VERSION
+};
+
+/* Header at the beginning of every event in the 'data' stream */
 #pragma pack(1)
 struct ctf_event_header {
   uint64_t timestamp;
-  uint32_t pid;
   uint32_t id;
 };
 
+/* The structs in this section are programmatic structs, i.e. they
+are used in the code but not directly emitted as is to the trace files. */
+
 struct event {
   struct ctf_event_header header;
   uint16_t  phase; /* for GC events */
   uint16_t  counter_kind; /* misc counter name */
   uint8_t  alloc_bucket; /* for alloc counters */
   uint64_t count; /* for misc counters */
+  uint8_t span_type; /* for user events */
```

Comment:
What is the `span_type` for a GC event?

Reply:
It was originally 0 (dummy value), but now `phase` and `span_type` have been combined.